File: 5.implementation/VT/redsun/src/main/webapp/static/js/services/BillLinesService.js


/**
 * Controller for ${entity.name}
 **/


app.factory('billLineService', function($http, $rootScope) {
	return {
	
		// Insert.
		add: function(billLine) {
			var urlServer = $rootScope.makePostURL("/billline/insert");
			return $http.post(urlServer, billLine);
		}
		
		// Update.
		, update: function(billLineId, billLine) {
			var urlServer = $rootScope.makePostURL("/billline/update/" + billLineId);
			return $http.put(urlServer, billLine);
		}
		
		// Delete.
		, delete: function(billLineId) {
			var urlServer = $rootScope.makePostURL("/billline/delete/" + billLineId);
			return $http.delete(urlServer);
		}
		
		// Get by Id.
		, getById: function(billLineId) {
			var url = $rootScope.makeGetURL("/billline/getbyid/" + billLineId);
			return $http.get(url);
		}
		
		// List for page and filter.
		, getsForPageAndFilter: function(itemsPerPage, pageNo, billLine) {
			var url = $rootScope.makeGetURL("/billline/list/page/filter/" + itemsPerPage + "/" + pageNo);
			return $http.get(url, { params: billLine });
		}
		
	}// return.
});

File: 5.implementation/redsun-service/src/main/java/com/redsun/service/service/BillLinesService.java
package com.redsun.service.service;

import com.redsun.service.entities.BillLines;
import com.redsun.service.entities.Result;

/**
 * BillLines Service interface
 */
public interface BillLinesService {
	
    // Insert.
	Result insert(final BillLines billLines) ;

    // Update.
	Result update(final BillLines billLines) ;

    // Delete.
	Result delete(final Integer id) ;

	// Get By Id.
	Result getById(final Integer id) ;

    // Exists.
	Result exists(final Integer id) ;

    // Count.
	Result count() ;

	// List for page and filter.
	Result listBillLinesForPageAndFilter(final int itemsPerPage, final int pageNo, final BillLines billLines);

}
